updater/config.gni

```diff
@@ -14,7 +14,19 @@ declare_args() {
   brave_enable_updater = false
 }
 
+declare_args() {
+  # It would be nice to declare build_omaha4 in the declare_args block above.
+  # But its default value depends on brave_enable_updater. In this case, GN
+  # requires us to create a separate declare_args block.
+  build_omaha4 = brave_enable_updater && !is_official_build
```

Comment:
this is a bit confusing. I assume omaha should be built in the "official" mode to benefit from all optimizations, but you're defining the default state like so to NOT build it in the official build. A comment on this would be nice.

the comment on how `declare_args` works is unnecessary.

Reply:
Should I rename `build_omaha4` to `build_omaha4_from_source`?